dokka-runners/dokka-gradle-plugin/src/main/kotlin/internal/PluginFeaturesService.kt

```diff
@@ -25,62 +27,102 @@ import java.util.*
+        val projectDisplayName: Property<String>
+
+        /** [Project.getProjectDir] - only used for log messages. */
+        val projectDirectory: Property<String>
     }
 
     /**
      * Designate this [BuildService] as 'primary', meaning it should log messages to users.
      * Non-primary services should not log messages.
      *
-     * Why? Because Gradle is buggy. Sometimes registering a BuildService fails.
-     * See https://github.com/gradle/gradle/issues/17559.
-     * If service registration fails then re-register the service, but with a distinct name
-     * (so it doesn't clash with the existing but inaccessible BuildService), and don't mark it as 'primary'.
+     * Why? Two reasons:
+     *
+     * 1. Sometimes registering a BuildService fails.
+     *   See https://github.com/gradle/gradle/issues/17559.
+     *   If service registration fails then re-register the service, but with a distinct name
+     *   (so it doesn't clash with the existing but inaccessible BuildService),
+     *   and don't mark it as 'primary' to avoid duplicate logging.
+     * 2. The method Gradle uses to generates accessors for pre-compiled script plugins
+     *   (see [PluginFeaturesService.Companion.configureParamsDuringAccessorsGeneration])
+     *   runs in a temporary project that is evaluated twice.
      *
      * @see org.jetbrains.dokka.gradle.internal.registerIfAbsent
      */
-    private val primaryService: Boolean get() = parameters.primaryService.getOrElse(false)
+    private val primaryService: Boolean
+        get() = parameters.primaryService.getOrElse(false)
 
     /**
      * Whether DGP should use V2 [org.jetbrains.dokka.gradle.DokkaBasePlugin].
      *
-     * Otherwise, fallback to V1 [org.jetbrains.dokka.gradle.DokkaClassicPlugin].
+     * @see pluginMode
      */
     internal val v2PluginEnabled: Boolean by lazy {
-        val v2PluginEnabled = parameters.v2PluginEnabled.getOrElse(false)
+        when (pluginMode) {
+            V1Enabled,
+                -> false
 
-        if (v2PluginEnabled) {
-            logV2PluginMessage()
-        } else {
-            logV1PluginMessage()
+            V2EnabledWithHelpers,
+            V2Enabled,
+                -> true
         }
+    }
+
+    /**
+     * Determines the behaviour of DGP.
+     *
+     * If no valid value is detected then use [PluginMode.Default].
+     */
+    private val pluginMode: PluginMode by lazy {
+        val parameterValue = parameters.pluginMode.getOrElse(PluginMode.Default.name)
+
+        val value = PluginMode.findOrNull(parameterValue)
+            ?: run {
+                logger.warn("Invalid value for $PLUGIN_MODE_FLAG. Got '$parameterValue' but expected one of: ${PluginMode.values.map { it.name }}")
+                PluginMode.Default
+            }
+
+        logger.info { "Detected PluginMode: $parameterValue/$value in ${parameters.projectDisplayName.orNull} ${parameters.projectDirectory.orNull}" }
```

Comment:
minor: if I remember correctly, in `info` mode there will be no mention of `logger.name` and so no mention of Dokka, am I right?
also, minor, but AFAIR Gradle logs project name if there are logs, so this additional information about project is redundant

```suggestion
        logger.info { "Detected Dokka Gradle Plugin mode: $parameterValue/$value in ${parameters.projectDisplayName.orNull} ${parameters.projectDirectory.orNull}" }
```

Reply:
Good idea, done.

> AFAIR Gradle logs project name if there are logs

I think not. This is what I see with `./gradlew dokkaGenerate --dry-run --debug`

```
2024-09-23T13:16:08.596+0200 [WARN] [org.jetbrains.dokka.gradle.internal.PluginFeaturesService] Dokka Gradle Plugin PluginMode: V2Enabled/V2Enabled in root project 'KmpCommonSourceSharedWithDependentsTest' /Users/dev/projects/jetbrains/dokka/dokka-runners/dokka-gradle-plugin/build/test-temp-dir/functional-tests/KmpCommonSourceSharedWithDependentsTest
```

Reply:
Ah, okay. Probably it happens only during configuration. I was meant this:
<img width="707" alt="image" src="https://github.com/user-attachments/assets/f8453e53-33ac-4fc5-9b43-b32de97689e0">